app/main/views/signup.py

```diff
@@ -66,14 +68,21 @@ def send_seller_signup_email():
 
 @main.route('/application/<int:id>', methods=['GET'])
 @main.route('/application/<int:id>/<path:step>', methods=['GET'])
+@login_required
 def application(id, step=None):
 
+    application = data_api_client.get_application(id)['application']
+
+    if not current_user.is_authenticated:
+        return current_app.login_manager.unauthorized()
+    if current_user.id != application['user_id']:
+        return current_app.login_manager.unauthorized()
+
     form = DmForm()
-    basename = url_for('.application', id=id, step=None)
-    rendered_component = render_component('bundles/ApplicantSignup/ApplicantSignupWidget.js',
-                                          {'form_options': {'csrf_token': form.csrf_token.current_token,
-                                                            'mode': 'edit',
-                                                            }, 'basename': basename})
+    application['basename'] = url_for('.application', id=id, step=None)
+    application['form_options'] = {'csrf_token': form.csrf_token.current_token}
```

Comment:
Latest release of utils makes this redundant, gets included automatically in `render_component`